### PR TITLE
refactor: remove deprecated usage of wagmi connectors

### PR DESCRIPTION
**Pull request by jxom**
`wagmi@^0.4` has deprecated the usage of passing a function to `connectors` in `createClient`. Now, it is recommended to just pass an array to `connectors`. [See here](https://wagmi.sh/docs/migrating-to-03#passing-a-function-to-createclient-connectors-has-now-been-deprecated). 

~This PR just changes the return type of `connectors` in `connectorsForWallets` from a function to an array. Everything should still work as intended.~

### CODE REVIEW

File: packages/rainbowkit/src/wallets/Wallet.ts [hunk 1]
```diff
@@ -40,9 +36,7 @@ export type Wallet<C extends Connector = Connector> = {
     browserExtension?: string;
     qrCode?: string;
   };
-  createConnector: (connectorArgs: {
-    chainId?: number;
-  }) => RainbowKitConnector<C>;
+  createConnector: () => RainbowKitConnector<C>;
```
Comment (jxom) [inline packages/rainbowkit/src/wallets/Wallet.ts:39]: idk if we should convert this from a function to a plain rainbowkit connector?

File: packages/rainbowkit/src/wallets/walletConnectors/coinbase/coinbase.ts [hunk 2]
```diff
@@ -23,18 +23,14 @@ export const coinbase = ({ appName, chains }: CoinbaseOptions): Wallet => {
...
 
-      const chain = chains.find(chain => chain.id === chainId);
-      const jsonRpcUrl = chain?.rpcUrls.default;
-
       const connector = new CoinbaseWalletConnector({
         chains,
         options: {
           appName,
           headlessMode: true,
-          jsonRpcUrl,
```
Comment (jxom) [inline packages/rainbowkit/src/wallets/walletConnectors/coinbase/coinbase.ts:37]: wagmi now looks after RPC URLs internally (from `chains`).

File: packages/rainbowkit/src/wallets/Wallet.ts (hunk 1 above)
Comment (peduarte) [reply packages/rainbowkit/src/wallets/Wallet.ts:39]: 🔥 

> idk if we should convert this from a function to a plain rainbowkit connector now that there are no params?

not sure, is a plain rainbowkit connector a simpler api?

---

just a reminder to update [docs props table](https://rainbowkit.vercel.app/docs/custom-wallets) when we settle on the api

CleanShot 2022-06-06 at 18 05 50@2x

Comment (jxom) [reply packages/rainbowkit/src/wallets/Wallet.ts:39]: Yeah it will be a simpler API! If we do go in this direction though, I wouldn't be sure on what to rename the property since there is already a `connector` property in the `RainbowKitConnector` type:

```diff
type RainbowKitConnector<C extends Connector = Connector> = {
  connector: C;
  ...
  };
};

export type Wallet<C extends Connector = Connector> = {
  ...
- createConnector: (connectorArgs: {
-   chainId?: number;
-  }) => RainbowKitConnector<C>;
+ connector: RainbowKitConnector<C>
};
```

Would having 2 `connector`s be weird?

**Comment by markdalgleish** [comment]
This looks great!

This PR will need a changeset — it's unlikely to break any consumers, but this is still technically a breaking change, so I think it'll need to be marked as a minor release so it goes out as v0.3.0.

**Comment by jxom** [comment]
@markdalgleish @peduarte – added some changesets and docs

File: packages/create-rainbowkit/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
 # @rainbow-me/create-rainbowkit
 
+## 0.0.4
```
Comment (markdalgleish) [inline packages/create-rainbowkit/CHANGELOG.md:3]: I'm not sure what happened, but it looks like your PR is picking up unrelated changes that are already in main.

File: site/data/docs/migrating-to-02.mdx [hunk 4]
```diff
@@ -1,12 +1,72 @@
...
 
-# Migrating to 0.2
+# Migrating to >=0.2
 
 ## Migrating RainbowKit
 
-RainbowKit has updated the `wagmi` peer dependency to `^0.4` which comes with a few [breaking changes](https://wagmi.sh/docs/migrating-to-03#04x-breaking-changes).
+### 0.3.x Breaking changes
+
+Removed the `chainId` parameter from `createConnector` on the `Wallet` type (Custom Wallets).
```
Comment (markdalgleish) [inline site/data/docs/migrating-to-02.mdx:11]: How about something like this, to make it clear that most consumers don't need to care:

```suggestion
Removed the `chainId` parameter from `createConnector` on the `Wallet` type.

**Note that all built-in wallets are using the new API. Most consumers will be unaffected. This change only affects consumers that have created/consumed custom wallets.**
```

File: .changeset/funny-tips-eat.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+---
+'@rainbow-me/rainbowkit': minor
+---
+
+**Breaking:** Removed the `chainId` parameter from `createConnector` on the `Wallet` type (Custom Wallets).
```
Comment (markdalgleish) [inline .changeset/funny-tips-eat.md:5]: Same suggestion as the migration guide on the docs site:

```suggestion
**Breaking:** Removed the `chainId` parameter from `createConnector` on the `Wallet` type.

**Note that all built-in wallets are using the new API. Most consumers will be unaffected. This change only affects consumers that have created/consumed custom wallets.**
```

File: .changeset/funny-tips-eat.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+If you previously derived RPC URLs from the `chainId` on `createConnector`, you can now remove that logic as `wagmi` now handles RPC URLs internally when used with `configureChains` .
```
Comment (markdalgleish) [inline .changeset/funny-tips-eat.md:7]:
```suggestion
If you previously derived RPC URLs from the `chainId` on `createConnector`, you can now remove that logic as `wagmi` now handles RPC URLs internally when used with `configureChains`.
```

File: site/data/docs/migrating-to-02.mdx [hunk 7]
```diff
@@ -1,12 +1,72 @@
...
+
+If you previously derived RPC URLs from the `chainId` on `createConnector`, you can now remove that logic as `wagmi` now handles RPC URLs internally when used with `configureChains` .
```
Comment (markdalgleish) [inline site/data/docs/migrating-to-02.mdx:13]:
```suggestion
If you previously derived RPC URLs from the `chainId` on `createConnector`, you can now remove that logic as `wagmi` now handles RPC URLs internally when used with `configureChains`.
```

File: .changeset/funny-tips-eat.md (hunk 5 above)
Comment (markdalgleish) [reply .changeset/funny-tips-eat.md:5]: I think this one got missed.

**Review by markdalgleish — APPROVED** [review]
Just one note about updating the changeset, but otherwise lgtm 👍

**Comment by markdalgleish** [comment]
@jxom Should we also remove usage of the `rpc` property in our WalletConnect wallets? I did this in my PR that also touches this code: https://github.com/rainbow-me/rainbowkit/pull/492

**Comment by jxom** [comment]
Yep, we can remove `rpc` now

File: .changeset/funny-tips-eat.md (hunk 5 above)
Comment (markdalgleish) [reply .changeset/funny-tips-eat.md:5]: Ah, sorry, I just noticed you updated this manually.

**Comment by jxom** [comment]
lgtm @markdalgleish
